platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/EventDescriptor.java

```diff
@@ -59,85 +64,96 @@ public ChatterEventDescriptor() {}
      * @param creator    the creator of the event
      * @param generation the age of an event, smaller is older
      */
-    public ChatterEventDescriptor(@NonNull final Hash hash, @NonNull final NodeId creator, final long generation) {
+    public EventDescriptor(@NonNull final Hash hash, @NonNull final NodeId creator, final long generation) {
         this.hash = Objects.requireNonNull(hash, "hash must not be null");
         this.creator = Objects.requireNonNull(creator, "creator must not be null");
         this.generation = generation;
 
-        hashCode = Objects.hash(hash, creator, generation);
+        hashCode = hash.hashCode();
```

Comment:
this does not need to be cached, I think we can just call `hash.hashCode()` directly